In [19]:
pip install pandas numpy matplotlib seaborn scipy

In [18]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# a) Load the dataset
df = pd.read_csv("C:/Users/sahaj/Downloads/property.csv")

# b) Inspect the first few rows of the dataset
print(df.head())

       Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
3  4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0     

In [20]:
# Check the structure of the dataset
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

1. For the suburb Altona, it is postulated that a typical property sells for $800,000. Use the data at hand to test this assumption. Is the typical property price really $800,000 or has it increased? Use a significance level of 5%. 

In [17]:
from scipy import stats

# Filter for Altona suburb
altona_data = df[df['Suburb'] == 'Altona']

# Remove missing prices
altona_prices = altona_data['Price'].dropna()

# Set the hypothesized price
hypothesized_price = 800000

# Perform one-sample t-test
t_stat, p_value = stats.ttest_1samp(altona_prices, hypothesized_price)

# Print results
print(f"Mean Price in Altona: {altona_prices.mean()}")
print(f"T-statistic: {t_stat}, P-value: {p_value}")

# Conclusion based on p-value
alpha = 0.05
if p_value < alpha:
    print("Reject the null hypothesis, the typical property price is different from $800,000.")
else:
    print("Fail to reject the null hypothesis, the typical property price is $800,000.")


Mean Price in Altona: 834830.4054054054
T-statistic: 1.0277020770199676, P-value: 0.307483271305555
Fail to reject the null hypothesis, the typical property price is $800,000.


Given a significance level of 5%, the p-value of 0.307483271305555 is greater than 0.05. This means we fail to reject the null hypothesis. Based on the statistical analysis, there is not enough evidence to conclude that the average property price in Altona has significantly increased from $800,000. 

2. For the year 2016, is there any difference in prices of properties sold in the summer months vs winter months? Consider months from October till March as winter months and the rest as summer months. Use a significance level of 5%. 

Comparison of Summer vs. Winter Property Prices (2016)

In [21]:
# Convert Date to datetime
df['Date'] = pd.to_datetime(df['Date'], format='%d/%m/%Y')

# Filter data for the year 2016
year_2016_data = df[df['Date'].dt.year == 2016]

# Define summer and winter months
summer_months = [4, 5, 6, 7, 8, 9]
winter_months = [10, 11, 12, 1, 2, 3]

# Filter data for summer and winter months
summer_data = year_2016_data[year_2016_data['Date'].dt.month.isin(summer_months)]
winter_data = year_2016_data[year_2016_data['Date'].dt.month.isin(winter_months)]

# Perform a two-sample t-test
t_stat, p_value = stats.ttest_ind(summer_data['Price'].dropna(), winter_data['Price'].dropna())

# Print results
print(f"Mean Summer Price: {summer_data['Price'].mean()}")
print(f"Mean Winter Price: {winter_data['Price'].mean()}")
print(f"T-statistic: {t_stat}, P-value: {p_value}")

# Conclusion
if p_value < alpha:
    print("There is a significant difference in property prices between summer and winter in 2016.")
else:
    print("There is no significant difference in property prices between summer and winter in 2016.")


Mean Summer Price: 1048054.7286917741
Mean Winter Price: 1116647.5917391304
T-statistic: -4.043386317851058, P-value: 5.3309767667631686e-05
There is a significant difference in property prices between summer and winter in 2016.


Given a significance level of 5%, the p-value of 5.3309767667631686e-05 is less than 0.05. This means we reject the null hypothesis.Based on the statistical analysis, there is a significant difference in property prices between summer and winter months in 2016

3. For the suburb Abbotsford, what is the probability that out of 10 properties sold, 3 will not have car parking? Use the column car in the dataset. Round off your answer to 3 decimal places. 
Probability of No Car Parking in Abbotsford

In [22]:
# Filter for Abbotsford suburb
abbotsford_data = df[df['Suburb'] == 'Abbotsford']

# Calculate probability of no car parking
no_car_parking_prob = len(abbotsford_data[abbotsford_data['Car'] == 0]) / len(abbotsford_data)

# Binomial probability for 3 out of 10 properties not having car parking
n = 10
k = 3
binom_prob = stats.binom.pmf(k, n, no_car_parking_prob)

print(f"Probability that 3 out of 10 properties will not have car parking: {binom_prob:.3f}")


Probability that 3 out of 10 properties will not have car parking: 0.260


In [ ]:
4. In the suburb Abbotsford, what are the chances of finding a property with 3 rooms? Round your answer to 3 decimal places. 
Probability of Finding a Property with 3 Rooms in Abbotsford

In [23]:
# Calculate probability of finding a property with 3 rooms
rooms_3_prob = len(abbotsford_data[abbotsford_data['Rooms'] == 3]) / len(abbotsford_data)

print(f"Probability of finding a property with 3 rooms: {rooms_3_prob:.3f}")


Probability of finding a property with 3 rooms: 0.357


5.In the suburb Abbotsford, what are the chances of finding a property with 2 bathrooms? Round your answer to 3 decimal places. 

 Probability of Finding a Property with 2 Bathrooms in Abbotsford

In [24]:
# Calculate probability of finding a property with 2 bathrooms
bathrooms_2_prob = len(abbotsford_data[abbotsford_data['Bathroom'] == 2]) / len(abbotsford_data)

print(f"Probability of finding a property with 2 bathrooms: {bathrooms_2_prob:.3f}")


Probability of finding a property with 2 bathrooms: 0.339
